Bellerophon/Bellerophon/Sources/BPManager.swift

```diff
@@ -124,39 +123,48 @@ public class BellerophonManager: NSObject {
         retryTimer = nil
     }
 
-    internal func performForceUpdate() {
-        delegate?.shouldForceUpdate()
-    }
-
     internal func handleError(error: NSError) {
-        delegate?.receivedError(error: error)
+        config.delegate?.receivedError(error: error)
     }
 
-    internal func displayKillSwitch() {
-        guard !killSwitchWindow.isKeyWindow else {
+    internal func displayWindowIfPossible(for event: BellerophonEvent) {
+        guard config.containsView(), let view = view(for: event) else {
             return
         }
+        currentEvent = event
+        view.frame = bellerophonWindow.bounds
+        config.allViews().forEach { $0.isHidden = true }
+        view.isHidden = false
+        config.delegate?.bellerophonWillEngage?(self, event: event)
+        bellerophonWindow.makeKeyAndVisible()
+    }
 
-        killSwitchView.frame = killSwitchWindow.bounds
-        delegate?.bellerophonWillEngage?(self)
-        killSwitchWindow.makeKeyAndVisible()
+    internal func view(for event: BellerophonEvent) -> UIView? {
```

Comment:
This function would not be necessary if we create an object that combines the event and the view together